### DISCUSSION TITLE
Как запустить проект

### DISCUSSION THREAD

**Discussion by ademaro** [makehtml/python-pikabu-tutorial#4] [Q&A]
1. Устанавливаем [Python 3](https://www.python.org/downloads/) если не установлено. В Windows python можно поставить из WondowsStore.
2. Клонируем репозиторий себе на компьютер и переходим в `develop` ветку: `git clone -b develop git@github.com:makehtml/python-pikabu-tutorial.git` (как работать с ветками написано [тут](https://github.com/makehtml/python-pikabu-tutorial/discussions/3))

6. Ставим окружение питона: `python3 -m venv .venv` (`python -m venv .venv`) _— это делается один раз_
7. Активируем окружение: `source .venv/bin/activate` (mac/linux/bash/zsh) или `.venv\Scripts\Activate.bat` (windows/cmd)
8. Ставим зависимости питона: `pip install -r requirements.txt` _— это делается один раз_
9. Переводим окружение для flask в режим разработки: `export FLASK_ENV=development` (bash) или `set FLASK_ENV=development` (cmd)

11. Запускаем проект: `flask run` 
12. Открываем браузер по адресу http://localhost:5000/

**UPDATE 01-12-2021**: Если возникают ошибки на этапе установки зависимости mysqlclient, нужно установить MySQL-сервер.
В Linux (Debian/Ubuntu): `sudo apt install mysql-server libmysqlclient-dev`, это делается один раз. 
В MacOS: `brew install mysql`, это делается один раз. 
Затем заново запустить установку зависимостей: `pip install -r requirements.txt`

**Comment by ANima93** [comment]
Если ошибка с mysql на windows?

**Reply by ademaro** [reply]
@ANima93 отпиши что за ошибки — скорее всего так же поставить mysql нужно (если именно mysql подключаешь)
